app/components/CreateAudio/index.tsx

```diff
@@ -10,79 +10,76 @@ import { Link } from '~/components/common/Link';
 import Feedback from '~/components/Feedback';
 import { ROUTES } from '~/routes/routes';
 import { VALID_GENRES } from '~/configs';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { audioSchema } from '~/hooks/useCreateEntity/schemas';
+import { useForm } from 'react-hook-form';
 
 const CreateAudio = () => {
-  const {
-    name,
-    releaseYear,
-    genre,
-    collectionID,
-    formValidity,
-    onChange,
-    signAndBroadcast,
-    broadcastStatus,
-  } = useCreateAudio();
+  const { signAndBroadcast, broadcastStatus } = useCreateAudio();
   const { collections } = useUserDiscography();
 
+  const { handleSubmit, register, formState } = useForm({
+    resolver: yupResolver(audioSchema),
+    mode: 'onBlur', // validate on blur
+    shouldFocusError: true, // focus input with error after submit
+    defaultValues: {
+      name: '',
+      releaseYear: '',
+      collectionID: '',
+      genre: 0,
+      files: null,
+      message: '',
+    },
+  });
+
+  const onSubmit = async (data: Record<string, any>) => {
+    await signAndBroadcast(data);
+  };
+
+  const errorMessage = formState.errors && Object.values(formState.errors)[0]?.message as string;
+  
   return (
-    <form className="component createAudio">
+    <form onSubmit={handleSubmit(onSubmit)} className='component createAudio'>
       <fieldset>
-        <Input
-          value={name}
-          onChange={onChange}
-          name="name"
-          placeholder="Enter name"
-          type="text"
-        />
-        <Input
-          value={releaseYear}
-          onChange={onChange}
-          name="releaseYear"
-          placeholder="Release year"
-          type="text"
-        />
+        <Input register={register} name='name' placeholder='Enter name' type='text' />
+        <Input register={register} name='releaseYear' placeholder='Release year' type='text' />
         <div className='collectionRow'>
           <Select
-            placeholder="Select a collection (Collection)"
-            name="collectionID"
+            placeholder='Select a collection (Collection)'
+            name='collectionID'
             options={collections}
-            value={collectionID}
-            onChange={onChange}
-          />
-          <Link
-            to={ROUTES.UPLOAD_COLLECTION}
-            icon="cross"
-            className='addCollection'
+            register={register}
```

Comment:
I'm not sure this is correct. I thin the Select element should be registered not the Link that redirects to the collection tab. Did you try creating an Audio? I assume this should crash or at least throw.